controller.rst

```diff
@@ -372,6 +371,23 @@ methods for storing and fetching things from the session::
 
 Stored attributes remain in the session for the remainder of that user's session.
 
+.. tip::
+
+    Every ``SessionInterface`` implementation is supported. If you have your
+    own implementation, type-hint this in the arguments instead.
+
+As a developer, you might prefer not to extend the ``Controller``. To use the
```

Comment:
This paragraph should be moved to the end of the next section ("Flash Messages") and can maybe be put in a `.. sidebar::` or `.. tip::` directive instead.